sdks/python/apache_beam/runners/portability/fn_api_runner/translations.py

```diff
@@ -149,15 +155,26 @@ def no_overlap(a, b):
     return (
         not consumer.forced_root and not self in consumer.must_follow and
         not self.is_runner_urn(context) and
-        not consumer.is_runner_urn(context) and
-        no_overlap(self.downstream_side_inputs, consumer.side_inputs()))
+        not consumer.is_runner_urn(context) and no_overlap(
+            set(self.downstream_side_inputs.keys()),
+            {i
+             for i, _, _ in consumer.side_inputs()}))
+
+  def _fuse_downstream_side_inputs(self, other):
+    res = dict(self.downstream_side_inputs)
+    for si, other_si_ids in other.downstream_side_inputs.items():
+      if si in res:
+        res[si] = union(res[si], other_si_ids)
```

Comment:
So this is actually a dict mapping to sets?

Reply:
woah this is a bug.
downstream side input is a dictionary mapping to dictionaries.

Dict[Output Pcollection, Dict[Side input ID, Access pattern]]
Where Side input ID is Tuple[consumer ptransform, input index]. Added appropriate annotations, and fixed the bug.